summerofcode/2023.md

```diff
@@ -26,6 +26,46 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
 
 ## Ideas
 
+### Armada
+
+#### Add Kubectl Plugin for Armada
+
+- Description: Armada is a multi-cluster batch scheduling platform for running high-throughput jobs.  Armada provides a CLI for interacting with jobs, queues and watching events. A common way that Kubernetes users interact with Kubernetes is via kubectl.  This project is to build a kubectl plugin that allows users to submit jobs to Armada using kubectl.
+- Expected outcomes:
+  - Publish a [Krew Plugin](https://krew.sigs.k8s.io/) for Armada
+  - Users can submit jobs, create queues, watch jobs etc using kubectl.
+- Recommend Skills: Go, Kubernetes, Kubectl
+- Mentor(s): Kevin Hannon, @kannon92
+- Expected project size: 100 Hours
+- Difficulty: Easy
+- Upstream Issue (URL): https://github.com/armadaproject/armada/issues/2120
+
+#### Build interfaces around Postgres for Armada
+
+- Description: Open source projects should not be hard coded to a particular Database. Armada currently only allows users to use Postgres.  This project is to build interfaces around our connections to Postgres so we can allow other databases.
+- Expected outcomes:
+  - A interface is created that allows Armada to interact with any SQL database without exposing implementation details of postgres
+  - With an interface, mocking and testing becomes much easier as you can implement mocks for the DB access.  Currently we require a postgres DB for unit testing.
+  - Test coverage could increase
+- Recommend Skills: Go, SQL
+- Mentor(s): Kevin Hannon, @kannon92
+- Expected project size: 200 Hours
+- Difficulty: Medium
+- Upstream Issue (URL): https://github.com/armadaproject/armada/issues/2121
+
+#### Create Deployment Guides for the Cloud Kubernetes Providers
+
+- Description: A common way that people deploy and use Kubernetes is to use a managed Kubernetes provider.  Armada is primarly used for on-prem Kubernetes clusters so our main documentation/experience is for running Kubernetes on Openstack.  This project will create deployment guides for AKS (Azure Kubernetes Service), GKE (Google Kubernetes Engine) and Openshift.
+- Expected outcomes:
+  - Deployment guide for Armada on AKS.
+  - Deployment guide for Armada on GKE.
+  - Deployment guide for Armada on Openshift.
```

Comment:
Although this is a great exercise for any contributor, my understanding of the GSoC rules is that the documentation only tasks are not good for GSoC.
Is there any coding needed here?

https://summerofcode.withgoogle.com/rules rule 1.26:

> “Project” means an open source coding project to be worked on by a GSoC Contributor as an individual. For the avoidance of doubt, Projects do not include projects for documentation only.

Reply:
Okay, I’ll remove this project idea.

Reply:
Removed.